docs/core_docs/docs/tutorials/qa_chat_history.ipynb

```diff
@@ -1,1438 +1,943 @@
+      "-----\n",
+      "\n"
+     ]
     }
+   ],
+   "source": [
+    "let inputs4 = { messages: [{ role: \"user\", content: \"Can you look up some common ways of doing it?\" }] };\n",
+    "\n",
+    "for await (\n",
+    "  const step of await graphWithMemory.stream(inputs4, threadConfig)\n",
+    ") {\n",
+    "    const lastMessage = step.messages[step.messages.length - 1];\n",
+    "    prettyPrint(lastMessage);\n",
+    "    console.log(\"-----\\n\");\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4bbbeef2-d9a1-4857-874f-9f3b5cc4eca9",
+   "metadata": {},
+   "source": [
+    "Note that the query generated by the model in the second question incorporates the conversational context.\n",
+    "\n",
+    "The [LangSmith](https://smith.langchain.com/public/c8b2c1ba-8c8b-47ab-b298-3502e0688711/r) trace is particularly informative here, as we can see exactly what messages are visible to our chat model at each step."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0ad23c71-3c99-4d9d-b494-9b7a08a557c0",
+   "metadata": {},
+   "source": [
+    "## Agents {#agents}\n",
+    "\n",
+    "[Agents](/docs/concepts/agents) leverage the reasoning capabilities of LLMs to make decisions during execution. Using agents allows you to offload additional discretion over the retrieval process. Although their behavior is less predictable than the above \"chain\", they are able to execute multiple retrieval steps in service of a query, or iterate on a single search.\n",
+    "\n",
+    "Below we assemble a minimal RAG agent. Using LangGraph's [pre-built ReAct agent constructor](https://langchain-ai.github.io/langgraph/how-tos/#langgraph.prebuilt.chat_agent_executor.create_react_agent), we can do this in one line.\n",
+    "\n",
+    "```{=mdx}\n",
+    ":::tip\n",
+    "\n",
+    "Check out LangGraph's [Agentic RAG](https://langchain-ai.github.io/langgraphjs/tutorials/rag/langgraph_agentic_rag/) tutorial for more advanced formulations.\n",
+    "\n",
+    ":::\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "6d519b4c-29e3-4e14-a5f8-178bd25a1728",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import { createReactAgent } from \"@langchain/langgraph/prebuilt\";\n",
+    "\n",
+    "const agent = createReactAgent({ llm: llm, tools: [retrieve] });"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7d8f8734-5dcf-4058-a532-11c8a7d0efae",
+   "metadata": {},
+   "source": [
+    "Let's inspect the graph:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "00cbb651-db1c-405d-905d-6e200c7169a7",
+   "metadata": {},
+   "source": [
+    "```javascript\n",
```

Comment:
nit: I usually use `ts`, just make sure highlighting looks proper in the final build